cmd/controlloop/controlloop.go

```diff
@@ -132,14 +139,21 @@ func newPodController(stopChannel chan struct{}) (*controlloop.PodController, er
 	const noResyncPeriod = 0
 	ipPoolInformerFactory := wbinformers.NewSharedInformerFactory(wbClientSet, noResyncPeriod)
 	netAttachDefInformerFactory := nadinformers.NewSharedInformerFactory(nadK8sClientSet, noResyncPeriod)
+
+	nodeName := os.Getenv(podControllerNodeNameEnvVariable)
+	logging.Debugf("Filtering pods with filter key '%s' and filter value '%s'", podControllerFilterKey, nodeName)
+	if nodeFound, err := nodeExists(k8sClientSet, nodeName); !nodeFound {
+		msg := fmt.Sprintf("WARNING: Could not find node with node name '%s'. The pod informer may not work correctly.",
+			nodeName)
+		if err != nil {
+			msg = fmt.Sprintf("%s Received an unexpected error message, err: %q", msg, err)
+		}
+		logging.Errorf(msg)
```

Comment:
I think we should flat out return an error here, making the pod crash loop if we can't find the node for which we want to "monitor" the workloads for.

Reply:
done